client/src/screens/SignIn/SignIn.jsx

```diff
@@ -1,35 +1,105 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
-import { Header } from '../../components/Header/Header.jsx';
-import { Form } from '../../components/Form/Form';
 // assets
 import { useDispatch } from 'react-redux';
 import { loginUser } from '../../services/redux/slices/uiActions.js';
-import { signInForm } from '../../services/formData.js';
+import '../SignUp/SignUp.scss';
+import store from '../../services/redux/store.js'
+import { GenericModal } from '../../components/Modal/GenericModal';
+import { handleChange } from '../../services/utils/formHandlers';
+import { SingleActionButton } from '../../components/Button/SingleActionButton';
+import { checkEmailAuth, signOut } from '../../services/api/users';
 
 export const SignIn = () => {
   const navigate = useNavigate();
   const dispatch = useDispatch();
+  const [noAccountError, setNoAccountError] = useState(false);
+  const [showModal, setShowModal] = useState(false);
   const [loginInfo, setLoginInfo] = useState({
     email: 'test@test.com',
     password: 'test',
   });
 
+  useEffect( async () => {
+    signOut();
+  },[])
+
   const handleSignIn = async (event) => {
     event.preventDefault();
-    dispatch(loginUser(loginInfo));
-    navigate('/');
+    if (!noAccountError) {
+      await dispatch(loginUser(loginInfo));
+      //check for user
+      const state = store.getState();
+      if (state.ui.user.email === loginInfo.email) {
+        navigate(`/users/${state.ui._id}/edit`);
+      } else {
+        setShowModal(true);
```

Comment:
we can consolidate the control flow & number of clicks the user makes by eliminating the need for a modal. I thought the text element that renders under the email input onBlur was subtle & perfect!